src/inference/enumerate.ad.js

```diff
@@ -67,21 +93,33 @@ module.exports = function(env) {
     // Find support of this distribution:
     if (dist.isContinuous || !dist.support) {
       console.error(dist);
-      throw new Error('Enumerate can only be used with distributions that have finite support.');
+      return 'Enumerate can only be used with distributions that have finite support.';
     }
     var supp = dist.support();
 
     // Check that support is non-empty
     if (supp.length === 0) {
       console.error(dist);
-      throw new Error('Enumerate encountered a distribution with empty support!');
+      return 'Enumerate encountered a distribution with empty support.';
     }
-
     return supp;
   };
 
   Enumerate.prototype.sample = function(store, k, a, dist) {
     var support = getSupport(dist);
+    if (_.isString(support)) {
```

Comment:
One way to make this work is to adjust `getSupport` to take a continuation. This is a drag, but it seems better than expecting every caller of `getSupport` to check the return value and call `this.error` as appropriate? Having the signature be `getSupport(dist, cont)` would be consistent with other similar code I think. (`cont` rather than `k` because this would be a JS continuation rather than a webppl one. e.g It doesn't take a `store` argument.)